src/docfx/build/xref/ExternalXrefMapLoader.cs

```diff
@@ -95,12 +99,41 @@ private static void LoadZipFile(Dictionary<string, Lazy<ExternalXrefSpec>> resul
                     var xrefMap = JsonUtility.Deserialize<XrefMapModel>(errors, reader, path);
                     foreach (var spec in xrefMap.References)
                     {
-                        result.TryAdd(spec.Uid, new Lazy<ExternalXrefSpec>(() => spec));
+                        result.TryAdd(spec.Uid, new Lazy<(string?, ExternalXrefSpec)>(() => (xrefMap.RepositoryName, spec)));
                     }
                 }
             }
         }
 
+        private static Dictionary<string, string> GetStringProperties(ReadOnlySpan<byte> content, params string[] propertyNames)
+        {
+            var properties = new Dictionary<string, string>();
+
+            if (propertyNames.Length == 0)
+            {
+                return properties;
+            }
+
+            var reader = new Utf8JsonReader(content, isFinalBlock: true, default);
+            while (reader.Read())
+            {
+                switch (reader.TokenType)
+                {
+                    case JsonTokenType.PropertyName:
```

Comment:
return only top level properties

Reply:
Maybe not,  I have confirmed it,  which will also return the non-root level properties.

Reply:
OK.